### ISSUE TITLE
Unable to run plugin after installing latest version

### ISSUE THREAD

**Issue by BurnyBoi** [poljar/weechat-matrix#95]
I just installed the latest version, and I'm seeing the following when I attempt to load the plugin:

```
2019-05-31 10:35:36		python: stdout/stderr (?): Traceback (most recent call last):
2019-05-31 10:35:36		python: stdout/stderr (?):   File "/home/user/.weechat/python/autoload/matrix.py", line 34, in <module>
2019-05-31 10:35:36		python: stdout/stderr (?):     from nio import RemoteProtocolError, RemoteTransportError, TransportType
2019-05-31 10:35:36		python: stdout/stderr (?):   File "/usr/lib/python3.7/site-packages/nio/__init__.py", line 2, in <module>
2019-05-31 10:35:36		python: stdout/stderr (?):     from .client import *
2019-05-31 10:35:36		python: stdout/stderr (?):   File "/usr/lib/python3.7/site-packages/nio/client/__init__.py", line 3, in <module>
2019-05-31 10:35:36		python: stdout/stderr (?):     from .base_client import Client, ClientConfig, logged_in, store_loaded
2019-05-31 10:35:36		python: stdout/stderr (?):   File "/usr/lib/python3.7/site-packages/nio/client/base_client.py", line 23, in <module>
2019-05-31 10:35:36		python: stdout/stderr (?):     from ..crypto import ENCRYPTION_ENABLED
2019-05-31 10:35:36		python: stdout/stderr (?):   File "/usr/lib/python3.7/site-packages/nio/crypto/__init__.py", line 4, in <module>
2019-05-31 10:35:36		python: stdout/stderr (?):     if package_installed("olm"):
2019-05-31 10:35:36		python: stdout/stderr (?):   File "/usr/lib/python3.7/site-packages/nio/_compat.py", line 27, in package_installed
2019-05-31 10:35:36		python: stdout/stderr (?):     spec = importlib.util.find_spec(package_name)
2019-05-31 10:35:36		python: stdout/stderr (?): AttributeError: module 'importlib' has no attribute 'util'
2019-05-31 10:35:36	=!=	python: unable to parse file "/home/user/.weechat/python/autoload/matrix.py"
```

**Comment by BurnyBoi** [comment]
When I ran pip to install the new requirements, it installed matrix-nio 0.3 and peewee 3.9.5. Let me know if any other info is needed

**Comment by poljar** [comment]
I am not sure what's going on here, python 3.7 should have [util](https://docs.python.org/3/library/importlib.html#importlib.util.find_spec).

**Comment by dkasak** [comment]
@BurnyBoi, can you check which path is `importlib` getting loaded from? Start up the Python 3.7 REPL and do

```
>>> import importlib
>>> importlib.util
<module 'importlib.util' from '/usr/lib/python3.7/importlib/util.py'>
```

**Comment by BurnyBoi** [comment]
Thanks for having a look! Here's what I got:

```
>>> import importlib
>>> importlib.util
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'importlib' has no attribute 'util'
```

**Comment by BurnyBoi** [comment]
I just confirmed I do have a /usr/lib/python3.7/importlib/util.py too.

**Comment by dkasak** [comment]
Uhh, sorry, my mistake. Obviously you don't have the `util` submodule for some reason, so you can't check its path.

Rather, do this:

```python
import importlib
print(importlib.__file__)

**Comment by BurnyBoi** [comment]
```
>>> import importlib
>>> print(importlib.__file__)
/usr/lib/python3.7/importlib/__init__.py
```

**Comment by dkasak** [comment]
Does this work?

```python
import importlib
import importlib.util
print(importlib.util.__file__)

**Comment by BurnyBoi** [comment]
Looks like it does:

```
>>> import importlib
>>> import importlib.util
>>> print(importlib.util.__file__)
/usr/lib/python3.7/importlib/util.py
```

**Comment by dkasak** [comment]
I guess this is the same problem as in [this SO question](https://stackoverflow.com/questions/39660934/error-when-using-importlib-util-to-check-for-library) then. Should be easy to fix.

**Comment by BurnyBoi** [comment]
Makes sense. Does this mean one of the files in weechat-matrix needs to have `import importlib.util` somewhere in it?

**Comment by dkasak** [comment]
It seems so. You can try this to see whether it fixes it for you: https://github.com/poljar/matrix-nio/pull/21

**Comment by BurnyBoi** [comment]
This is weird, I'm looking at my `/usr/lib/python3.7/site-packages/nio/_compat.py` file, and it doesn't look like what you have:

```
import sys

if sys.version_info >= (3, 5):
    import importlib

    # The imp module is deprecated by importlib but importlib doesn't have the
    # find_spec function on python2. Use the imp module for py2 until we
    # deprecate python2 support.

    def package_installed(package_name):
        spec = importlib.util.find_spec(package_name)

        if spec is None:
            return False
        else:
            return True
else:
    import imp

    def package_installed(package_name):
        try:
            imp.find_module(package_name)
            return True
        except ImportError:
            return False
```

I just have the version of matrix-nio 0.3 from pip

**Comment by BurnyBoi** [comment]
Okay, one more update; I just manually changed my _compat.py, and it looks like both `import importlib.util` AND `from importlib import util` will work just fine. The problem is that, as you can see above, mine has neither. It seems like the version that's in pip is just too out of date compared to what you have in github.

**Comment by mmsenxxx** [comment]
Just installed, followed instructions. End result:

> python: stdout/stderr (?): Traceback (most recent call last):
03:13:59      weechat     | python: stdout/stderr (?):   File "/home/xxxxx/.weechat/python/matrix.py", line 25, in <module>
03:13:59      weechat     | python: stdout/stderr (?):     from builtins import str
03:13:59      weechat     | python: stdout/stderr (?): ImportError: No module named builtins
03:13:59      weechat =!= | python: unable to parse file "/home/xxxxx/.weechat/python/matrix.py"

**Comment by poljar** [comment]
You seem to be missing the [future](https://pypi.org/project/future/) python package. I don't think this is in any way related to the issue you are commenting on.

**Comment by BurnyBoi** [comment]
Sorry for the delayed response. This fell off my radar. I had future installed the entire time during my testing. I'm still using the _combat.py where I added the import for util myself. I'm not sure if matrix-nio in pip has been updated to include the import, but it's still my understanding that was the issue.

**Comment by poljar** [comment]
It hasen't been updated on pip/pypi yet but will so this week. Thanks for responding. I will close this once nio has been updated on pypi.

**Comment by poljar** [comment]
Fixed in nio 0.4.1.
